sdk/appconfiguration/azure-data-appconfiguration/src/configuration_client.cpp

```diff
@@ -0,0 +1,2316 @@
+  response.m_accept = accept;
+  response.m_options = options;
+  {
+    auto const& responseBody = rawResponse->GetBody();
+    if (responseBody.size() > 0)
+    {
+      auto const jsonRoot
+          = Core::Json::_internal::json::parse(responseBody.begin(), responseBody.end());
+      if (jsonRoot.contains("@nextLink") && !jsonRoot["@nextLink"].is_null())
+      {
+        response.NextPageToken = jsonRoot["@nextLink"].get<std::string>();
+      }
+
+      if (jsonRoot.contains("items"))
+      {
+        response.Items = std::vector<Key>{};
+
+        for (auto const& jsonItem : jsonRoot["items"])
+        {
+          Key vectorItem{};
+
+          vectorItem.Name = jsonItem["name"].get<std::string>();
+
+          response.Items.Value().emplace_back(std::move(vectorItem));
+        }
+      }
+    }
+  }
+
+  response.SyncToken = rawResponse->GetHeaders().at("Sync-Token");
+  response.ContentType = GetKeysResponseContentType(rawResponse->GetHeaders().at("Content-Type"));
+
+  response.RawResponse = std::move(rawResponse);
+
+  return response;
+}
+
+Azure::Response<CheckKeysResult> ConfigurationClient::CheckKeys(
+    CheckKeysOptions const& options,
+    Core::Context const& context)
+{
+  auto url = m_url;
+  url.AppendPath("keys");
+
+  url.AppendQueryParameter("api-version", Core::Url::Encode(m_apiVersion));
+  url.AppendQueryParameter("name", Core::Url::Encode(options.Name));
+  url.AppendQueryParameter("After", Core::Url::Encode(options.After));
+  Core::Http::Request request(Core::Http::HttpMethod::Head, url);
+
+  request.SetHeader("Accept", "application/json");
+
+  request.SetHeader("Sync-Token", options.SyncToken);
+  request.SetHeader("Accept-Datetime", options.AcceptDatetime);
+  request.SetHeader("x-ms-client-request-id", options.XMsClientRequestId.ToString());
+  auto rawResponse = m_pipeline->Send(request, context);
+  auto const httpStatusCode = rawResponse->GetStatusCode();
+
+  if (httpStatusCode != Core::Http::HttpStatusCode::Ok)
+  {
+    throw Core::RequestFailedException(rawResponse);
+  }
+
+  CheckKeysResult response{};
+  response.SyncToken = rawResponse->GetHeaders().at("Sync-Token");
+
+  return Response<CheckKeysResult>(std::move(response), std::move(rawResponse));
+}
+
+GetKeyValuesPagedResponse ConfigurationClient::GetKeyValues(
+    std::string const& accept,
+    GetKeyValuesOptions const& options,
+    Core::Context const& context)
+{
+  auto url = m_url;
+  url.AppendPath("kv");
+
+  url.AppendQueryParameter("api-version", Core::Url::Encode(m_apiVersion));
+  url.AppendQueryParameter("key", Core::Url::Encode(options.Key));
+  url.AppendQueryParameter("label", Core::Url::Encode(options.Label));
+  url.AppendQueryParameter("After", Core::Url::Encode(options.After));
+  {
+    std::string valueStr;
+    for (size_t i = 0; i < options.Select.size(); ++i)
+    {
+      if (i != 0)
+      {
+        valueStr += ',';
+      }
+
+      valueStr += Core::Url::Encode(options.Select[i].ToString());
```

Comment:
I suspect it was because of the `i != 0` clause for appending a `,` to `valueStr` in between.

VS Copilot suggested the following:
```C++
    std::string valueStr;

    for (auto const& item : options.Select)
    {
      if (!valueStr.empty())
      {
        valueStr += ',';
      }

      valueStr += Core::Url::Encode(item.ToString());
    }
```

Reply:
https://github.com/Azure/autorest.cpp/issues/430